### DISCUSSION TITLE
[Firebase messaging] How do you handle the app icon badge number in iOS for push notifications?

### DISCUSSION THREAD

**Discussion by downhamd** [firebase/flutterfire#10462] [Q&A]
When you receive push notifications on iOS the red badge number appears on the app icon, you normally have to add a package such as [flutter_app_badger](https://pub.dev/packages/flutter_app_badger) or [flutter_dynamic_icon](https://pub.dev/packages/flutter_dynamic_icon) to remove the app icon badge (or decrement it by 1). Is this correct?

I have tried these packages in my Flutter app, and although I can now remove the app icon badge with one line of code, I now have the problem of the app icon badge NO LONGER appearing when I receive push notifications. Could somebody kindly explain to me why I am no longer getting a badge on the app icon when I receive push notifications after I added these packages? And if I am going in the wrong direction, could somebody also explain how you normally handle the red app icon badge number in iOS when you implement firebase messaging in your Flutter app?

Thank you very much.

**Comment by petrnymsa** [comment]
Hi @downhamd, did you solve it? We have same issue

**Comment by DylanCross** [comment]
I also haven't found a good way to do this. It doesn't seem like there is a consensus solution, although this is 100% a requirement to be able to use push notifications with flutter / firebase.
